### DISCUSSION TITLE
Stretching on Hi-res mobile displays

### DISCUSSION THREAD

**Discussion by k0mrad** [cables-gl/cables_docs#474] [! Welcome - readme first!]
Having issues when a patch is embedded and viewed on a hi-res device. In my case: Samsung Galaxy S10. Brave Browsers. There are no issues when viewing this on desktop/laptop and in landscape mode on hi-res devices. 

In portrait mode, the canvas is stretched/squished. 

The issue is present in both cases: 

https://cdpn.io/pen/debug/BaPxwWe/c1250ad61215e9422e04cae451122325
https://codepen.io/komrad/pen/gOjeddj/9368485a3709fb6fb714508baa6591a6

**Comment by k0mrad** [comment]
Any ideas? I have found that there is no issue in iOS, or when the cable is iframed on the same Android device. Its only when the Canvas is rendered full screen as an embed in HTML.

**Comment by pandrr** [comment]
i guess the mixture of `"glCanvasResizeToWindow": true,` when using   `object-fit:contain;`
will not work, i dont know why it happens only on some browsers...

**Comment by steam0r** [comment]
this is what the two link slook like for me on android firefox, doesnt look stretched, does it?

![Screenshot_20230206-114618](https://user-images.githubusercontent.com/151271/216952323-ac0fdffb-ccd7-474d-865b-b203ea69fcbb.png)
![Screenshot_20230206-114556](https://user-images.githubusercontent.com/151271/216952346-bde42044-9e24-479d-8472-a3f71a6a4526.png)

**Comment by k0mrad** [comment]
i'm not sure what happened, but about a week ago it stopped being stretched for me too @pandrr @steam0r

**Comment by pandrr** [comment]
we did not change anything

**Reply by k0mrad** [reply]
i spoke too soon. they are still stretched... removing object-fit doesn't fix that.

**Comment by pandrr** [comment]
did you try just exporting it ? i blame codepen css

**Reply by k0mrad** [reply]
Everything i do is exported. I then host it on s3 and launch with JS.

**Reply by steam0r** [reply]
since you reopened this: can you provide a link to your exported version and screenshots, thanks! preferrably just the cables exported version with no additional modifications. maybe you can use the netlify or github export feature of cables. then test there.

**Reply by k0mrad** [reply]
![Screenshot_20230215-052730_Brave](https://user-images.githubusercontent.com/12103441/219045099-aea3efa7-2607-4679-b1ed-b710cd1f6784.jpg)

https://cdpn.io/pen/debug/gOjeddj/9368485a3709fb6fb714508baa6591a6

**Reply by k0mrad** [reply]
The debug view on codepen is the one that doesn't force it into an iframe and this is where the issue happens.

**Comment by steam0r** [comment]
again: please provide an export that does not use some external css or html, we can't debug integrations into whatever environments, sorry.

**Reply by k0mrad** [reply]
Sure, here it is as just an export 
https://s3.us-west-2.amazonaws.com/files.thedigitalpanda.com/test/cables/scroll/index.html

the only css is the following:


    html, body {
                width: 100%;
                height: 300vh;
                background: #EBE4E4;
                margin:0;
                padding:0;
                min-height:100vh;
                }

                canvas {
                display: block;
                width:100vw!important;
                height:100vh!important;
                position:fixed;
                left:0;
                top:0;
                pointer-events:none;
                }

**Reply by steam0r** [reply]
and do you now have a link to the patch somewhere? we REALLY cannot help you by just looking at things from the outside...

still looks like the above screenshot on my google pixel in chrome and firefox...are you maybe somehow handling landscape and portrait differently? guessing...

**Reply by k0mrad** [reply]
thats my bad as I assumed you would be looking at the source: 

https://s3.us-west-2.amazonaws.com/files.thedigitalpanda.com/test/cables/scroll/patch.js

**Reply by steam0r** [reply]
i mean your cables project, that js is minified and basically unreadable, as you might see

**Reply by k0mrad** [reply]
my bad again, thanks for clarifying (this process is new to me)
https://cables.gl/p/kdgeCn

**Reply by steam0r** [reply]
this is how this is looking for me on hires-retina macbook m1 and mobile hires:

<img width="507" alt="Bildschirm­foto 2023-02-15 um 19 22 59" src="https://user-images.githubusercontent.com/151271/219119190-cce510e3-df96-4cbd-9f58-459916839ab6.png">

<img width="380" alt="Bildschirm­foto 2023-02-15 um 19 23 13" src="https://user-images.githubusercontent.com/151271/219119123-a1fe7811-a08c-489e-a371-3fc261f8b1c9.png">

**Reply by steam0r** [reply]
also the remote viewer is a good way to test this on mobile, while patching
